src/test/java/com/final_assignment/UserServiceTest.java

```diff
@@ -23,25 +22,64 @@ class UserServiceTest {
     private UserService userService;
 
     @Mock
-    private UserService userMapper;
+    private UserMapper userMapper;
+
+    private List<User> mockUsers;
+
+    @BeforeEach
+    void setUp() {
+        List<User> mockUsers = Arrays.asList(
+                new User(1, "yamada", "yamada@example.com"),
+                new User(1, "kobashi", "kobashi@example.com"),
+                new User(1, "akiyama", "akiyama@example.com"),
+                new User(1, "misawa", "misawa@example.com"),
+                new User(1, "takayama", "takayama@example.com")
+                );
+    }
     
     @Test
-    void findByNameStartingWith() {
+    void 全レコードを取得する事() {
+        doReturn(mockUsers).when(userMapper).findAll();
+        List<User> actual = userService.findByNameStartingWith(null);
+        assertThat(actual).isEqualTo(mockUsers);
     }
 
     @Test
-    void 存在するユーザーのIDを指定したときに正常にユーザーが返されること() {
-            doReturn(Optional.of(new User(1, "yamada", "yamada@example.com"))).when(userMapper).findName(1);
-            User actual = userService.findName(1);
-            assertThat(actual).isEqualTo(new User(1, "yamada", "yamada@example.com"));
+    void クエリ文字列を指定してレコードを取得できること() {
+        doReturn(List.of(new User(4, "misawa", "misawa@example.com"))).when(userMapper).findByNameStartingWith("m");
+        List<User> actual = userService.findByNameStartingWith("m");
+        assertThat(actual).isEqualTo(List.of(new User(4, "misawa", "misawa@example.com")));
     }
 
     @Test
-    void insert() {
+    void 指定したIDのレコードを取得できること() {
+        doReturn(Optional.of(new User(1, "yamada", "yamada@example.com"))).when(userMapper).findName(1);
```

Comment:
findNameでなくfindByIdじゃないでしょうか？
```suggestion
        doReturn(Optional.of(new User(1, "yamada", "yamada@example.com"))).when(userMapper).findById(1);
```